bundle/ScheduledVisibility/Handler/Location.php

```diff
@@ -0,0 +1,150 @@
+    ) {}
+
+    public function hide(Content $content): void
+    {
+        try {
+            if ($this->isHidden($content)) {
+                return;
+            }
+        } catch (BadStateException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Content with id #%d is in the wrong state: %s',
+                    $content->getId(),
+                    $e->getMessage(),
+                ),
+            );
+
+            return;
+        }
+
+        if (method_exists($this->contentService, 'hideContent')) {
+            $this->repository->sudo(
+                fn () => $this->contentService->hideContent($content->getContentInfo()),
+            );
+
+            return;
+        }
+
+        try {
+            $locations = $this->repository->sudo(
+                fn () => $this->locationService->loadLocations($content->getContentInfo()),
+            );
+        } catch (BadStateException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Content with id #%d is in the wrong state: %s',
+                    $content->getId(),
+                    $e->getMessage(),
+                ),
+            );
+
+            return;
+        }
+
+        foreach ($locations as $location) {
+            $this->repository->sudo(
+                fn () => $this->locationService->hideLocation($location),
+            );
+        }
+    }
+
+    public function reveal(Content $content): void
+    {
+        try {
+            if (!$this->isHidden($content)) {
+                return;
+            }
+        } catch (BadStateException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Content with id #%d is in the wrong state: %s',
+                    $content->getId(),
+                    $e->getMessage(),
+                ),
+            );
+
+            return;
+        }
+
+        if (method_exists($this->contentService, 'revealContent')) {
+            $this->repository->sudo(
+                fn () => $this->contentService->revealContent($content->getContentInfo()),
+            );
+
+            return;
+        }
+
+        try {
+            $locations = $this->repository->sudo(
+                fn () => $this->locationService->loadLocations($content->getContentInfo()),
+            );
+        } catch (BadStateException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Content with id #%d is in the wrong state: %s',
+                    $content->getId(),
+                    $e->getMessage(),
+                ),
+            );
+
+            return;
+        }
+
+        foreach ($locations as $location) {
+            $this->repository->sudo(
+                fn () => $this->locationService->unhideLocation($location),
+            );
+        }
+    }
+
+    /**
+     * @throws BadStateException
+     */
+    public function isHidden(Content $content): bool
```

Comment:
The problem here is if you try to use this method in reverse, to check if the given `$content` is visible with `!$handler->isHidden()`. In that case it will not work correctly as it will return `true` when `Content` is visible, but `Locations` are not visible.

I think we have to check on both, and we also need to update visibility on both.

It should be properly documented as well.

Reply:
fixed in 16f5f6f28154ef69948f2965fad53d9240e65d7d